packages/lib/src/vis/shared/Annotation.tsx

```diff
@@ -10,18 +11,30 @@ interface Props extends HTMLAttributes<HTMLDivElement> {
   x: number;
   y: number;
   scaleOnZoom?: boolean;
+  center?: boolean;
 }
 
 function Annotation(props: Props) {
-  const { x, y, scaleOnZoom, children, style, ...divProps } = props;
+  const { x, y, scaleOnZoom, center, children, style, ...divProps } = props;
 
   const camera = useThree((state) => state.camera);
+  useFrameRendering();
 
   const { dataToWorld, worldToHtml } = useAxisSystemContext();
   const htmlPt = worldToHtml(dataToWorld(new Vector2(x, y)));
 
-  useFrameRendering();
-
+  if ((center || scaleOnZoom) && style?.transform) {
+    throw new Error(
+      'Annotation with `center` and/or `scaleOnZoom` cannot have its own `transform`'
+    );
+  }
+
+  const transforms = [
+    center ? 'translate(-50%, -50%)' : undefined,
+    scaleOnZoom
+      ? `scale(${1 / camera.scale.x}, ${1 / camera.scale.y})`
+      : undefined,
+  ];
```

Comment:
You could use template strings instead to avoid the `filter(isDefined).join` below.
  
```ts
  const transform = `${center ? 'translate(-50%, -50%)' : ''}
  ${scaleOnZoom ? ` scale(${1 / camera.scale.x}, ${1 / camera.scale.y})` : ''}`;
```

Feel free to reject this suggestion.

Reply:
My OCD has an issue with that leading space when `scaleOnZoom` is `true` and `center` is `false` (which could also be a trailing space with the opposite conditions). :rofl: I could add a `trim()`though :thinking:

Reply:
I kept an array but replaced `undefined` with empty strings and used `join(' ').trim()`.